Log/lambda_function.py

```diff
@@ -62,28 +58,44 @@ def lambda_handler(event, context):
     #Add custom tags here by adding new value with the following format "key1:value1, key2:value2"  - might be subject to modifications
     metadata[DD_CUSTOM_TAGS] = "functionname:" + context.function_name+ ",memorysize:"+ context.memory_limit_in_mb
 
+    try:
+        logs = generate_logs(event)
+        for log in logs:
+            s = safe_submit_log(s, log)
+    except Exception as e:
+        print('Uncaught exception: {}'.format(str(e)))
+    finally:
+        s.close()
+
+def connect_to_datadog(host, port):
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    s = ssl.wrap_socket(s)
+    s.connect((host, port))
+    return s
 
+def generate_logs(event):
     try:
         # Route to the corresponding parser
         event_type = parse_event_type(event)
-
         if event_type == "s3":
-            logs = s3_handler(s, event)
-
+            logs = s3_handler(event)
         elif event_type == "awslogs":
-            logs = awslogs_handler(s, event)
-
-        for log in logs:
-            send_entry(s, log)
-
+            logs = awslogs_handler(event)
     except Exception as e:
         # Logs through the socket the error
         err_message = 'Error parsing the object. Exception: {}'.format(str(e))
-        send_entry(s, err_message)
-        raise e
-    finally:
-        s.close()
+        logs = [err_message]
+    return logs
 
+def safe_submit_log(s, log):
+    try:
+        send_entry(s, log)
+    except Exception as e:
+        err_message = 'Error sending the log line. Exception: {}'.format(str(e))
```

Comment:
I would print the log here.

Reply:
We're actually sending it to datadog!

Reply:
I meant doing something like:
```
err_message = 'Exception: {} occured for log line: {}'.format(str(e), log)
```

Reply:
Feedback from customer: we should not log when an error happened but we were able to retry